packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -1781,6 +1781,18 @@ export default class NewVm extends BaseComponent {
             <SectionContent>
               <Item label={_('secureBoot')}>
                 <Toggle onChange={this._toggleState('secureBoot')} value={secureBoot} />
+                {secureBoot && pool.allowSecureBoot === '' && (
```

Comment:
Reading the issue, I think we also need to check that the selected boot firmware is "uefi"?